.github/workflows/go.yml

```diff
@@ -14,17 +14,17 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        go: [1.21]
```

Comment:
As only one version is tested, removed the matrix for this and leverage the go.mod file to run with the version as defined in there, preventing duplication of the go version in various places that ran out of sync.